e2e_report_test.go

```diff
@@ -55,6 +55,78 @@ func TestE2EReport_Names(t *testing.T) {
 	assert.Contains(t, reporter.recorded, "RequestExample")
 }
 
+func TestE2EReport_Formatting(t *testing.T) {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/example-path", func(w http.ResponseWriter, r *http.Request) {
+	})
+	mux.HandleFunc("/number", func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte("8"))
+	})
+	mux.HandleFunc("/array", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		_, _ = w.Write([]byte("[7, 3, 9, 12, 4]"))
+	})
+
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	reporter := &recordingReporter{}
+
+	e := WithConfig(Config{
+		BaseURL:  server.URL,
+		Reporter: reporter,
+	})
+
+	e.GET("/example-path").
+		Expect().
+		Status(400)
+
+	t.Logf("%s", reporter.recorded)
+
+	assert.Contains(t, reporter.recorded, "/example-path")
+
+	assert.Contains(t, reporter.recorded, "unexpected http status value")
+	assert.Contains(t, reporter.recorded, "400 Bad Request")
+	assert.Contains(t, reporter.recorded, "200 OK")
+
+	reporter.recorded = ""
```

Comment:
Maybe extract each such section into subtest?

Reply:
Also, I suggest to match all of the values in each section / subtest: Actual, Expected, Reference (if present), Delta (if present). I think it makes it easier to read the test - we see all expectations at once.